src/dash-playlist-loader.js

```diff
@@ -0,0 +1,173 @@
+import { EventTarget } from 'video.js';
+import mpdParser from 'mpd-parser';
+import {
+  setupMediaPlaylists,
+  resolveMediaGroupUris
+} from './playlist-loader';
+
+export default class DashPlaylistLoader extends EventTarget {
+  // DashPlaylistLoader must accept either a src url or a playlist because subsequent
+  // playlist loader setups from media groups will expect to be able to pass a playlist
+  // (since there aren't external URLs to media playlists with DASH)
+  constructor(srcUrlOrPlaylist, hls, withCredentials) {
+    super();
+
+    this.hls_ = hls;
+    this.withCredentials = withCredentials;
+
+    if (!srcUrlOrPlaylist) {
+      throw new Error('A non-empty playlist URL or playlist is required');
+    }
+
+    // initialize the loader state
+    if (typeof srcUrlOrPlaylist === 'string') {
+      this.srcUrl = srcUrlOrPlaylist;
+      this.state = 'HAVE_NOTHING';
+      return;
+    }
+
+    this.state = 'HAVE_METADATA';
+    this.started = true;
+    // we only should have one, so select it
+    this.media(srcUrlOrPlaylist);
+    this.trigger('loadedplaylist');
+  }
+
+  dispose() {
+    this.stopRequest();
+  }
+
+  stopRequest() {
+    if (this.request) {
+      const oldRequest = this.request;
+
+      this.request = null;
+      oldRequest.onreadystatechange = null;
+      oldRequest.abort();
+    }
+  }
+
+  media(playlist) {
+    // getter
+    if (!playlist) {
+      return this.media_;
+    }
+
+    // setter
+    if (this.state === 'HAVE_NOTHING') {
+      throw new Error('Cannot switch media playlist from ' + this.state);
+    }
+
+    // find the playlist object if the target playlist has been specified by URI
+    if (typeof playlist === 'string') {
+      if (!this.master.playlists[playlist]) {
+        throw new Error('Unknown playlist URI: ' + playlist);
+      }
+      playlist = this.master.playlists[playlist];
+    }
+
+    const mediaChange = !this.media_ || playlist.uri !== this.media_.uri;
+
+    this.state = 'HAVE_METADATA';
```

Comment:
Are these states doc'ed anywhere? Might be nice to have them doc'ed in some form, as it might make debugging easier (particularly if there is a sequence to these events).

Reply:
these states mimic the PlaylistLoader states
https://github.com/videojs/http-streaming/blob/master/docs/arch.md

Reply:
Ah nice, this is what I was looking for.